pennylane/wires.py

```diff
@@ -30,30 +30,36 @@ def _process(wires):
         # if input is already a Wires object, just return its wire tuple
         return wires.labels
 
-    elif isinstance(wires, (Number, str)):
+    if isinstance(wires, (Number, str)):
         # interpret as a single wire
         return (wires,)
 
-    elif hasattr(wires, "shape") and wires.shape == tuple():
+    if hasattr(wires, "shape") and wires.shape == tuple():
         # Scalar NumPy array
         return (wires.item(),)
 
-    elif isinstance(wires, Iterable) and all(isinstance(w, Wires) for w in wires):
-        # if the elements are themselves Wires objects, merge them to a new one
-        return tuple(w for wires_ in wires for w in wires_.tolist())
-
-    elif isinstance(wires, Iterable) and all(
-        isinstance(w, (str, Number)) or (getattr(w, "shape", None) == tuple()) for w in wires
-    ):
-        # if the elements are strings or numbers, turn iterable into tuple
-        return tuple([w.item() if isinstance(w, np.ndarray) else w for w in wires])
+    while isinstance(wires, Iterable):
+        # wires is an iterable, presumably of Wires, Number or scalar NumPy array instances
+        if all(isinstance(w, Wires) for w in wires):
+            # if the elements are themselves Wires objects, merge them to a new one
+            merged = tuple(w for wires_ in wires for w in wires_.tolist())
+        elif all(
+            isinstance(w, (str, Number)) or (getattr(w, "shape", None) == tuple()) for w in wires
+        ):
+            # if the elements are strings or numbers, turn iterable into tuple
+            merged = tuple([w.item() if isinstance(w, np.ndarray) else w for w in wires])
+        else:
+            break
+        # check that all wires are unique
+        if len(set(merged)) != len(merged):
+            raise WireError("Wires must be unique; got {}.".format(merged))
```

Comment:
Aside from the restructuring of the if statements above, these two lines are the major change, right?

Reply:
Yes, this check was moved to `_process`.

Reply:
Very clever, how such a small change makes such a difference. Thanks!